src/platform/datapath_raw_socket.c

```diff
@@ -79,20 +76,91 @@ CxPlatTryAddSocket(
     _In_ CXPLAT_SOCKET* Socket
     )
 {
-    BOOLEAN Success = TRUE;
+    int Result;
+    BOOLEAN Success = FALSE;
     CXPLAT_HASHTABLE_LOOKUP_CONTEXT Context;
     CXPLAT_HASHTABLE_ENTRY* Entry;
 
+    //
+    // Get (and reserve) a transport layer port from tcpip.sys by binding an auxiliary socket.
+    //
+
+    Socket->AuxSocket =
+        WSASocketW(
+            AF_INET6,
```

Comment:
A comment explaining why a IPv6 socket is used would help.

Reply:
Can we use the normal `socket` function to keep this code xplat?